dawn/examples/tutorial/README.md

```diff
@@ -0,0 +1,106 @@
+
+```
+globals {
+  double dx;  //grid spacing
+};
+
+stencil laplacian_stencil {
+  storage_ij out_field;
+  storage_ij in_field;
+  Do() {
+    vertical_region(k_start, k_end) {
+	    out_field[i,j] = (-4*in_field + in_field[i+1] + in_field[i-1] + in_field[j-1] + in_field[j+1])/(dx*dx);
+    }
+  }
+};
+```
+
+This code defines two fields which will serve as the arguments to the stencil. The variable `dx` is the grid spacing and is read-only (during the stencil run), which is modelled as a global in **GTClang**. Observe how close the actual Laplacian stencil is to the numerical formula (c.f. for example [wikipedia](https://en.wikipedia.org/wiki/Finite_difference#Finite_difference_in_several_variables)), which close to no boiler plate. Save the stencil as `laplacian_stencil.cpp`.
+
+For the purpose of this tutorial we are going to use the `C++-naive` backend. To compile the stencil use:
+```
+./gtclang -backend=c++-naive laplacian_stencil.cpp -o laplacian_stencil_cxx_naive.cpp
+```
+
+## Writing and Compiling the Driver Code
+
+**GTClang** output a c++11-compliant source file. This code reads in a field `in_field`, applies the stencil, and writes the result into `out_field`. To use this, we need a driver. For the purpose of this exercise we are going initialize `in_field` to a wave function `in(x,y) = sin(x)*sin(y)`, since the Laplacian of this is the same wave again, but with inverted phase and twice the amplitude, and thus easy to check. The driver code is located in `laplacian_driver.cpp` and should be straightforward. The actual stencil launch is just one line:
+
+```
+dawn_generated::cxxnaive::laplacian_stencil laplacian_naive(dom, out, in);
+laplacian_naive.set_dx(dx);
+laplacian_naive.run(out, in);   //launch stencil
+```
+
+the run method could now be called in a time loop, for example to simulate diffusion. To facilitate the compilation, a `CMakeLists.txt` file has been provided. To compile the code:
+
+```
+mkdir build && cd build && cmake .. && make
+```
+
+This will place an executable called `laplacian_driver` in the tutorial directory. When run, two `vtk` files will be written. Those can be viewed using (ParaView)[https://www.paraview.org/]. `in.vtk` shows the initial conditions. If `out.vtk` is loaded on top, the inversion of phase and twicefold increase in amplitude can clearly be seen, as well as the halos around the domain, which would overlap with a "neighboring" MPI rank in practical implementations.
+
+<img src="img/in.png" width="425"/> <img src="img/out.png" width="425"/> 
+
+## Use Python to generate SIR 
+
+Another option to use **dawn** without having to rely on the **GTClang** DSL is to use the Python interface provided. As a preliminary step, a config file that was generated during the **dawn** install step needs to be copied and the proper `PYTHONPATH` needs to be set. Assuming you built dawn using the bundle this is:
+
+```
+cd <path/to/dawn>/dawn/examples/
+cp ../bundle/install/examples/python/config.py .
```

Comment:
instead of copying the config file into the src directory we could also go to the install/examples directory to run it there - as it was intended.

It is up for discussion if this should be in build/examples or if install is good enough, but that is for another PR

Reply:
This is the only point I did not address in the clean up. At some point, we should discuss installing the tutorial.

Reply:
yeah, I'm fine with that